homeassistant/components/binary_sensor/strings.json

```diff
@@ -302,7 +302,7 @@
     }
   },
   "device_class": {
```

Comment:
I think that all these need to be capitalized but I am not 100% sure.

Reply:
Binary sensor is also the only one using this.

Reply:
If we capitalize them, the documentation will need to clarify the difference between GUI and YAML configuration.

https://www.home-assistant.io/integrations/binary_sensor/

Reply:
> I think that all these need to be capitalized but I am not 100% sure.

They should not I think? They are used by the frontend rendering the logbook, and are often used mid-sentence. All other use-cases have been removed and replaced by the entity attribute translations.

IMHO, this whole list should be refactored out and removed.

Reply:
I added some auto casing logic into the frontend for binary sensor device classes in the logbook just a day or so ago. So that when a "heat" device class sensor switches on, it says "Detected heat" with a lower case "h" in EN, but keeps it capitalized for example in German.

Reply:
@spacegaier right, so it could use the translations provided by the entity component instead?

Reply:
Yes, those are now already used by the frontend for the binary sensors in the logbook as of two days ago (there might be other places, where we still need to add the auto casing as well):

https://github.com/home-assistant/frontend/blob/4d4bd3e912bd55531ae7186dc6c2c974208ddf60/src/data/logbook.ts#L365

Reply:
Nice ok, let's keep this out of scope for this PR in that case 👍